### ISSUE TITLE
Cupertino Date picker with Reachability in IOS accessibility on

### ISSUE THREAD

**Issue by StrawHt** [flutter/flutter#51810]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: 

[bug 2.zip](https://github.com/flutter/flutter/files/4278733/bug.2.zip)

 1. create an application that shows CupertinoDatePicker
2. turn on reachability in iOS accessbility
3. try to pick a date -->

**Expected results:** Be able to pick a date without triggering reachability

**Actual results:**  Reachability triggers and is unable to pick date properly

<details>
  <summary>Logs</summary>

```
[logs.txt](https://github.com/flutter/flutter/files/4278718/logs.txt)

```

```
Jasons-MacBook-Pro:datepicker jasonzheng$ flutter analyze
Analyzing datepicker...                                                 
No issues found! (ran in 2.7s)
```

```
Jasons-MacBook-Pro:datepicker jasonzheng$ flutter doctor -v
[✓] Flutter (Channel beta, v1.14.6, on Mac OS X 10.14.6 18G103, locale en-US)
    • Flutter version 1.14.6 at /Volumes/Data/Sources/flutter
    • Framework revision fabeb2a16f (5 weeks ago), 2020-01-28 07:56:51 -0800
    • Engine revision c4229bfbba
    • Dart version 2.8.0 (build 2.8.0-dev.5.0 fc3af737c7)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/jasonzheng/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.42.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (1 available)
    • iPhone 11 Pro • 64C37970-1AC3-4F52-A335-46701F492CFA • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!
```

</details>

**Comment by TrueNight** [comment]
I have same issue

**Comment by rawmind** [comment]
Reproduced on IOS 13.3

**Comment by TahaTesser** [comment]
Hi @StrawHt 
i was used `CupertinoDatePicker`on iOS 13.4  with  Reachability enabled, seen no issue

### Code Sample
```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      debugShowCheckedModeBanner: false,
      title: 'Cupertino App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  Future<void> showCupertinoDatePicker(BuildContext context) async {
    await showCupertinoModalPopup<void>(
      context: context,
      builder: (context) {
        return Container(
          height: 200,
          child: CupertinoTheme(
            data: CupertinoThemeData(
              textTheme: CupertinoTextThemeData(),
            ),
            child: CupertinoDatePicker(
              backgroundColor: Colors.white,
              mode: CupertinoDatePickerMode.date,
              initialDateTime: DateTime.now(),
              onDateTimeChanged: (date) {
                print(date);
              },
            ),
          ),
        );
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
        navigationBar: CupertinoNavigationBar(
          middle: Text('Cupertino App Bar'),
        ),
        child: Center(
          child: Container(
              padding: EdgeInsets.only(left: 25.0, right: 25.0),
              height: 45.0,
              child: GestureDetector(
                onTap: () async {
                  await showCupertinoDatePicker(context);
                },
                child: Container(
                  child: Text(DateTime.now().toIso8601String()),
                ),
              )),
        ));
  }
}
```

flutter doctor -v
```
[✓] Flutter (Channel dev, v1.16.2, on Mac OS X 10.15.4 19E266, locale en-GB)
    • Flutter version 1.16.2 at /Users/taha/Code/flutter_dev
    • Framework revision 58cad787d8 (3 days ago), 2020-03-23 08:33:41 -0700
    • Engine revision 2d42c74a34
    • Dart version 2.8.0 (build 2.8.0-dev.16.0 ba8baa46b4)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (4 available)
    • iPhone 11  • 41EB2A85-7EB4-4FD2-8978-FBBE2A681D74 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-4 (simulator)
    • macOS      • macOS                                • darwin-x64     • Mac
      OS X 10.15.4 19E266
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 80.0.3987.149
    • Web Server • web-server                           • web-javascript •
      Flutter Tools

• No issues found!
```

**Comment by rawmind** [comment]
It's still reproducible on 13.4. I've just checked it after OS upgrade. Take a note the bug is reproducible only on real device not emulator.

**Comment by AlexFox63** [comment]
When will this issue be fixed?

**Comment by maxlapides** [comment]
@jmagman @TahaTesser This issue is still affecting our iOS users. Are there plans for the Flutter team to work on this issue, or should we pursue an alternative solution?

**Comment by danagbemava-nc** [comment]
Reproduces on the latest versions of flutter.

https://user-images.githubusercontent.com/88313112/209930863-14e12655-fc04-4e54-91cd-a1a5c5d4c091.MP4

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.1 22C65 darwin-arm, locale en-GB)
    • Flutter version 3.3.10 on channel stable at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (2 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (3 available)
    • Nexus (mobile)  • 00008020-001875E83A38002E • ios            • iOS 16.2 20C65
    • macOS (desktop) • macos                     • darwin-arm64   • macOS 13.1 22C65 darwin-arm
    • Chrome (web)    • chrome                    • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[!] Flutter (Channel master, 3.7.0-13.0.pre.109, on macOS 13.1 22C65 darwin-arm64, locale en-GB)
    • Flutter version 3.7.0-13.0.pre.109 on channel master at /Users/nexus/dev/sdks/flutters
    ! Warning: `flutter` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bdefebcf73 (4 hours ago), 2022-12-29 00:46:32 -0500
    • Engine revision 9b534a5f88
    • Dart version 3.0.0 (build 3.0.0-71.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (3 available)
    • Nexus (mobile)  • 00008020-001875E83A38002E • ios            • iOS 16.2 20C65
    • macOS (desktop) • macos                     • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)    • chrome                    • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

**Comment by jmagman** [comment]
cc @hangyujin @MitchellGoodwin
